core/spec/models/spree/preferences/preferable_spec.rb

```diff
@@ -266,6 +266,44 @@ def self.allowed_admin_form_preference_types
         expect(@a.preferences[:product_attributes]).to eq({ id: 1, name: 2 })
       end
     end
+
+    context "converts encrypted_string preferences to encrypted values" do
+      it "with string, encryption key provided as option" do
+        A.preference :secret, :encrypted_string,
+                     encryption_key: 'VkYp3s6v9y$B?E(H+MbQeThWmZq4t7w!'
+
+        @a.set_preference(:secret, 'secret_client_id')
+        expect(@a.get_preference(:secret)).to eq('secret_client_id')
+        expect(@a.preferences[:secret]).not_to eq('secret_client_id')
+      end
+
+      it "with string, encryption key provided as env variable" do
+        ENV['SOLIDUS_PREFERENCES_MASTER_KEY'] = 'VkYp3s6v9y$B?E(H+MbQeThWmZq4t7w!'
```

Comment:
I am not sure but I guess that this ENV variable setting may leak in subsequent specs run after this one 🤔

Reply:
You were totally right, I've stubbed the request to ENV to avoid this.